server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -96,7 +99,19 @@ public IngestService(ClusterService clusterService, ThreadPool threadPool,
                 threadPool.getThreadContext(), threadPool::relativeTimeInMillis,
                 (delay, command) -> threadPool.schedule(
                     command, TimeValue.timeValueMillis(delay), ThreadPool.Names.GENERIC
-                ), this
+                ), this, index -> {
```

Comment:
I think we will need to do some alias resolution here as it appears that this call requires the concrete index ?

Reply:
I see now you do it in the test processor ... but maybe move the logic here to have one place for failures of that kind ?

Reply:
:+1: yeah it makes sense to move the index expression resolution to here.

The `Function<Index, Engine.Searcher>` would then change into `Function<String, Engine.Searcher>`.